common/processor/processor/src/main/java/io/helidon/common/processor/ElementInfoPredicates.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.processor;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import io.helidon.common.types.TypeName;
+import io.helidon.common.types.TypeNames;
+import io.helidon.common.types.TypeValues;
+import io.helidon.common.types.TypedElementInfo;
+
+/**
+ * Commonly used predicates to filter typed element info.
+ *
+ * @see io.helidon.common.types.TypedElementInfo
+ * @see io.helidon.common.types.TypeInfo#elementInfo()
+ */
+public final class ElementInfoPredicates {
+    /**
+     * Predicate for method element kind.
+     *
+     * @param element typed element info to test
+     * @return whether the element represents a method
+     */
+    public static boolean isMethod(TypedElementInfo element) {
+        return TypeValues.KIND_METHOD.equals(element.elementTypeKind());
+    }
+
+    /**
+     * Predicate for static modifier.
+     *
+     * @param element typed element info to test
+     * @return whether the element has static modifier
+     */
+    public static boolean isStatic(TypedElementInfo element) {
+        return element.modifiers().contains(TypeValues.MODIFIER_STATIC);
+    }
+
+    /**
+     * Predicate for private modifier.
+     *
+     * @param element typed element info to test
+     * @return whether the element has private modifier
+     */
+    public static boolean isPrivate(TypedElementInfo element) {
```

Comment:
there are parts of the code where we check for package private - we should add that method as well.

Reply:
Feel free to do that when you need it. I have only needed these methods in the code I was refactoring. I am sure there will be other useful predicates that we repeat...